<?xml version="1.0"?>
<launch>	
	 <param name="/use_sim_time" value="true" />
	<node name="explore_node" pkg="exp_devel" type="explore_node"  output="screen">
		<param name="scan_topic" value="/scan_fix" />
		<param name="laser_clockwise" value="true" />
		<param name="clicked_point" value="/clicked_point" />
		<param name="map_frame" value="/map" />
		<param name="map_topic" value="/map" />
		<param name="odom_frame" value="/odom" />
		<param name="use_odom" value="true" />
		<param name="odom_topic" value="/odometry/filtered" />
		<param name="vertices" value="4" />
		<param name="clear_dist" value="1.5" />
		<param name="scan_segment" value="0.2" />
		<param name="max_rot_vel" value="2" /> <!--max vel rot-->
		<param name="max_lin_vel" value="6" /> <!--max vel lin-->
		<param name="sensor_range" value="6" />
		<param name="inflation" value="1" /> <!--map inflation should same as the max radius of robot-->
		<param name="robot_span" value="1" /> <!--max robot dimension (length, width or radius) -->
		<!--innner resolution of secondary map. Highly Influences speed of algorithm. Lower the value in high performing PC -->
		<param name="resolution" value="0.2" /> 	
	</node>
	
	<node name="laser_hot_fix" pkg="exp_devel" type="laser_hot_fix"  output="screen">
		<param name="max_range" value="6.0" />
		<param name="sub_scan_topic" value="scan" />
		<param name="pub_scan_topic" value="scan_fix" />
	</node>
	 
	<include file="$(find husky_navigation)/launch/move_base.launch" />
	
	<include file="$(find husky_navigation)/launch/gmapping.launch" />	
	
	<include file="$(find measure_distance)/launch/measure_distance.launch" >
		 <arg name="odom_topic" value="odometry_filtered"/>
		 <arg name="feedback_topic" value="move_base/feedback"/>
	</include>

</launch>